### PR TITLE
[dune] Have only one rule calling configure

### PR DESCRIPTION
**Pull request by proux01**
There were two after #9867

### CODE REVIEW

File: configure.ml [hunk 1]
```diff
@@ -1216,7 +1216,7 @@ let write_dune_c_flags f =
   close_out o;
   Unix.chmod f 0o444
 
-let _ = try write_dune_c_flags "kernel/byterun/dune.c_flags" with _ -> ()
+let _ = try write_dune_c_flags "config/dune.c_flags" with _ -> ()
```
Comment (SkySkimmer) [inline configure.ml]: Why do we catch exceptions here btw?

Comment (proux01) [reply configure.ml]: Well spotted, it is useless. Removed.

**Review by ejgallego — APPROVED** [review]
LGTM, thanks!

We could maybe make the file shorter by actually placing the `c_flags` field inside the library stanza:
```lisp
(library
 (name byterun)
 (synopsis "Coq's Kernel Abstract Reduction Machine [C implementation]")
 (public_name coq.vm)
 (c_flags (:include ...))
 (c_names coq_fix_code coq_memory coq_values coq_interp))
```

**Comment by proux01** [comment]
done

**Comment by ejgallego** [comment]
Thanks a lot, will merge as soon the CI passes.

**Pull request merged**